notifications_utils/logging.py

```diff
@@ -16,46 +16,60 @@
 logger = logging.getLogger(__name__)
 
 
+def build_log_line(extra_fields):
+    fields = []
+    if 'service_id' in extra_fields:
+        fields.append(str(extra_fields.get('service_id')))
+    fields += [extra_fields.get('method'), extra_fields.get('url'), str(extra_fields.get('status'))]
+    if 'time_taken' in extra_fields:
+        fields.append(extra_fields.get('time_taken'))
+    return ' '.join(fields)
+
+
+def build_statsd_line(extra_fields):
+    fields = []
+    if 'service_id' in extra_fields:
+        fields.append(str(extra_fields.get('service_id')))
+    fields += [extra_fields.get('method'), extra_fields.get('endpoint'), str(extra_fields.get('status'))]
+    return '.'.join(fields)
+
+
 def init_app(app, statsd_client=None):
     app.config.setdefault('NOTIFY_LOG_LEVEL', 'INFO')
     app.config.setdefault('NOTIFY_APP_NAME', 'none')
     app.config.setdefault('NOTIFY_LOG_PATH', './log/application.log')
 
     @app.after_request
     def after_request(response):
-
         extra_fields = {
             'method': request.method,
             'url': request.url,
             'status': response.status_code
         }
 
+        if 'service_id' in g:
+            extra_fields.update({
+                'service_id': g.service_id
+            })
+
         if 'start' in g:
             time_taken = monotonic() - g.start
             extra_fields.update({
                 'time_taken': "%.5f" % time_taken
             })
 
-        if statsd_client and 'endpoint' in g:
-            statsd_client.incr(
-                '{method}.{endpoint}.{status_code}'.format(
-                    method=request.method, endpoint=g.endpoint, status_code=response.status_code
-                )
-            )
+        if 'endpoint' in g:
+            extra_fields.update({
+                'endpoint': g.endpoint
+            })
+
+        if statsd_client:
+            statsd_client.incr(build_statsd_line(extra_fields))
 
             if 'time_taken' in extra_fields:
-                statsd_client.timing(
-                    '{method}.{endpoint}.{status_code}'.format(
-                        method=request.method,
-                        endpoint=g.endpoint,
-                        status_code=response.status_code),
-                    time_taken
-                )
-
-        if 'time_taken' in extra_fields:
-            current_app.logger.info('{method} {url} {status} {time_taken}', extra=extra_fields)
-        else:
-            current_app.logger.info('{method} {url} {status}', extra=extra_fields)
+                statsd_client.timing(build_statsd_line(extra_fields), time_taken)
```

Comment:
DRY: you can build_statsd_line just once in a variable above rather than calling it twice